### PR TITLE
[CircleCI] Added Codecov.io integration

### PR DESCRIPTION
**Pull request by insomniacslk**
Codecov.io is a service that provides code coverage information.

Signed-off-by: Andrea Barberio <insomniac@slackware.it>

### CODE REVIEW

**Comment by insomniacslk** [comment]
I'm experimenting integration with codecov - this is not yet ready to review

**Comment by rjoleary** [comment]
Is this the same code coverage report as on Fiano? If so, it gets my approval.

Coverage has been difficult to measure in some places of u-root due tests which fork and tests which use qemu. I expect Codecov.io to have the same issues. Despite this, as with Fiano, the reports would still do more good than harm.

File: .circleci/config.yml [hunk 1]
```diff
@@ -112,6 +115,8 @@ jobs:
       - run:
           name: Test coverage
           command: go test -cover ./cmds/... ./pkg/...
+      - codecov/upload:
+          file: {{ coverage_report_filepath }}
```
Comment (rjoleary) [inline .circleci/config.yml]: It looks like yaml has a parse error here. I don't know yaml enough to help. Perhaps look use this snippet from fiano? https://github.com/linuxboot/fiano/blob/master/.circleci/config.yml#L108

Comment (insomniacslk) [reply .circleci/config.yml]: yeah, I tried to do as suggested on the codecov docs but it yields this error :( I'll go with the curl command instead

**Comment by insomniacslk** [comment]
Yeah it's the same coverage done in Fiano, and yeah, same issues with proper coverage measurements, but as you also say it's still better than nothing

**Comment by rjoleary** [comment]
Thanks! If this is ready, can you remove the "[WIP]" from the title?

**Review by rjoleary — APPROVED** [review]

**Comment by MedicineYeh** [comment]
Following the step of setting up the codecov. I added the environment variable in CIrcle CI project settings.
I think I'm not able to update this PR so I file another PR separately.
https://github.com/u-root/u-root/pull/1929

**Pull request closed without merging**